src/raftstore/store/peer.rs

```diff
@@ -791,6 +786,17 @@ impl Peer {
         send_msg.set_from_peer(from_peer);
         send_msg.set_to_peer(to_peer);
 
+        if self.get_store().is_initialized() &&
+           (msg_type == MessageType::MsgRequestVote ||
+            // the peer has not been known to this leader, it may exist or not.
```

Comment:
This comment is not quite useful. Comments occur in this code block should give a plain explanation on the purpose of setting region start key and end key in these messages. Something like:

```If a leader doesn't have the commit index info for a peer, there could be two cases:
1. That peer already exists but has not establish communication with leader yet
2. That peer is added newly due to member change or region split, but it's not created yet
For case 2, the region start key and end key are attached in RequestVote and Heartbeat message for the  store of that peer to check whether to create a new peer for member change when receiving these messages, or just to wait for a pending region split to perform later.
```